### PR TITLE
feat(button-group): initial design guidance

### PR DESCRIPTION
**Pull request by andyfeds**
Added initial guidance for button group.
[CDE-2345](https://vmw-jira.broadcom.net/browse/CDE-2345)-Create MD file for Button Group component.

### CODE REVIEW

File: 1004-button-group-design-guidance.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-15
+updated: 2024-10-22
+---
+
+# Button Group Component
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:8]:
```suggestion
# Button Group Design Guidance
```

File: 1004-button-group-design-guidance.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+The [button group](https://clarity.design/documentation/button-group) component is used to create collections of similar type of action buttons.
+
+## Guidance
+
+Button groups follow normal [button](https://guidance.clarity.design/1003) design guidelines. All button options regarding classes and sizes apply to button groups.
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:14]:
```suggestion
Button groups follow normal [button](/1003) design guidelines. All button options regarding classes and sizes apply to button groups.
```

Local link url should be relative.

File: 1004-button-group-design-guidance.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+### Button Group Types
+
+- **Must** use `solid` button groups to direct the user's attention to the primary actions the application is suggesting the user take.
+- **Must** use `outline` button groups to indicate a set of secondary actions or to reduce the visual noise on a page.
+- **Must** use `flat` button groups for a set of tertiary actions.
+- **Should** use `flat` button groups in cards and above datagrid components.
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:21]:
```suggestion
- **Must** use solid button groups to direct the user's attention to the primary actions the application is suggesting the user take.
- **Must** use outline button groups to indicate a set of secondary actions or to reduce the visual noise on a page.
- **Must** use flat button groups for a set of tertiary actions.
- **Should** use flat button groups in cards and above datagrid components.
```

No need for inline code.

File: 1004-button-group-design-guidance.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+- **May** use `solid`, `outline` and `flat` buttons in normal (default) and small (compact) sizes.
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:22]:
```suggestion
- **May** use all button types in normal (default) and small sizes.
```

Simplify wording and don't introduce an alternative word for the small button size.

File: 1004-button-group-design-guidance.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+### Button Group Arrangement
+
+- **Must** arrange primary actions on the left, followed by secondary actions, and warning actions as the last button in a group.
+- **Must not** mix differently styled buttons together, such as solid and outlined.
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:27]:
```suggestion
- **Must not** mix differently-styled buttons together, such as solid and outlined.
```

File: 1004-button-group-design-guidance.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+### Icon Button Group
+
+- **Must** use 'icon button groups' in their normal size (36px). This makes them easier to recognize and to click.
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:31]:
```suggestion
- **Must** use icon button groups in their normal size (36px). This makes them easier to recognize and to click.
```

File: 1004-button-group-design-guidance.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+- **Should** have text along with the icon if you have the space. This helps users understand the action.
+- **Should** start icon buttons with icons and follow with text.
+
+### Button Group Overflow
+
+- **Must** use overflow when a button group is larger than its containing space, or to preserve space.
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:37]:
```suggestion
- **Must** use overflow when a button group is larger than its containing space or to preserve space.
```

File: 1004-button-group-design-guidance.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+- Button groups with text and icons, or just icons, **should** have text show in the overflow dropdown menu.
+
+### Multiple Button Groups
+
+- **Must** start with the group of primary actions followed by the group of secondary actions.
+- **Should** be used when you want similar actions to be grouped together and separated from others.
+
+### Checkbox Button Groups
+
+- **Should** be used when a small list of options can be selected from, similar to the [checkbox](https://guidance.clarity.design/1006) component.
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:48]:
```suggestion
- **Should** be used when a small list of options can be selected from, similar to the [checkbox](/1006) component.
```

relative link url

File: 1004-button-group-design-guidance.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+- **Should** use option labels that are 1-3 short words.
+
+### Radio Button Groups
+
+- **Should** be used when selecting one option from a small list of options, similar to the [radio](https://guidance.clarity.design/1021) component.
```
Comment (kevinbuhmann) [inline 1004-button-group-design-guidance.md:53]:
```suggestion
- **Should** be used when selecting one option from a small list of options, similar to the [radio](1021) component.
```

relative link url

File: 1004-button-group-design-guidance.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-15
+updated: 2024-10-22
+---
+
+# Button Group Design Guidance
+
+The [button group](https://clarity.design/documentation/button-group) component is used to create collections of similar type of action buttons.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:10]:
```suggestion
The [button group](https://clarity.design/documentation/button-group) component is used to create collections of similar action buttons, allowing users to easily identify and perform related tasks within the interface.
```

File: 1004-button-group-design-guidance.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+## Guidance
+
+Button groups follow normal [button](/1003) design guidelines. All button options regarding classes and sizes apply to button groups.
+
+### Button Group Types
+
+- **Must** use solid button groups to direct the user's attention to the primary actions the application is suggesting the user take.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:18]:
```suggestion
- Design teams **must** use solid button groups to direct the user's attention to the primary actions the application suggests. This enhances visibility and encourages users to engage with the most important tasks.
```

File: 1004-button-group-design-guidance.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+- **Must** use outline button groups to indicate a set of secondary actions or to reduce the visual noise on a page.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:19]:
```suggestion
- Design teams **must** use outline button groups to indicate a set of secondary actions or to minimize visual noise on a page. This helps prioritize primary actions while still providing users with alternative options.
```

File: 1004-button-group-design-guidance.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+- **Must** use flat button groups for a set of tertiary actions.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:20]:
```suggestion
- Design teams **must** use flat button groups for tertiary actions. This helps distinguish these less critical actions from primary and secondary options, ensuring a clear hierarchy of user interactions.
```

File: 1004-button-group-design-guidance.md [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+- **Should** use flat button groups in cards and above datagrid components.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:21]:
```suggestion
- Design teams **should** use flat button groups in cards and above datagrid components. This placement enhances accessibility and keeps related actions easily discoverable for users.
```

File: 1004-button-group-design-guidance.md [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+- **May** use all button types in normal (default) and small sizes.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:22]:
```suggestion
- Design teams **may** use all button types in both normal (default) and small sizes. This flexibility allows for better alignment with the overall layout and user interface requirements.
```

File: 1004-button-group-design-guidance.md [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+
+### Button Group Arrangement
+
+- **Must** arrange primary actions on the left, followed by secondary actions, and warning actions as the last button in a group.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:26]:
```suggestion
- Design teams **must** arrange primary actions on the left, followed by secondary actions, and place warning actions as the last button in a group. This order provides a clear hierarchy and helps users quickly identify the most important actions.
```

File: 1004-button-group-design-guidance.md [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+- **Must not** mix differently-styled buttons together, such as solid and outlined.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:27]:
```suggestion
- Design teams **must not** mix differently styled buttons, such as solid and outlined, in the same button group. Maintaining a consistent style helps prevent confusion and ensures a cohesive user experience.
```

File: 1004-button-group-design-guidance.md [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+
+### Icon Button Group
+
+- **Must** use icon button groups in their normal size (36px). This makes them easier to recognize and to click.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:31]:
```suggestion
- Design teams **must** use icon button groups in their normal size (36px). This size enhances recognition and clickability, improving overall user interaction with the buttons.
```

File: 1004-button-group-design-guidance.md [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+- **Should** have text along with the icon if you have the space. This helps users understand the action.
+- **Should** start icon buttons with icons and follow with text.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:33]:
```suggestion
- Design teams **should** position icons before the text in icon buttons. This layout creates a natural flow for users, making it easier to identify actions quickly and intuitively.
```

File: 1004-button-group-design-guidance.md [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+
+### Button Group Overflow
+
+- **Must** use overflow when a button group is larger than its containing space or to preserve space.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:37]:
```suggestion
- Design teams **must** use overflow for button groups that exceed their containing space or to conserve space. This ensures that all actions remain accessible while maintaining a tidy interface.
```

File: 1004-button-group-design-guidance.md [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+- The overflow button **should** be the last position to the right of the button group.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:38]:
```suggestion
- Design teams **should** place the overflow button at the far right of the button group. This positioning makes it easy for users to find additional actions without disrupting the visual hierarchy.
```

File: 1004-button-group-design-guidance.md [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+- Button groups with text and icons, or just icons, **must** have text show in the overflow dropdown menu.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:39]:
```suggestion
- Design teams **must** ensure that button groups containing text and icons, or just icons, display text in the overflow dropdown menu. This practice enhances clarity and helps users understand the actions available to them.
```

File: 1004-button-group-design-guidance.md [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+
+### Multiple Button Groups
+
+- **Must** start with the group of primary actions followed by the group of secondary actions.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:43]:
```suggestion
- Design teams **must** arrange button groups to start with primary actions, followed by secondary actions. This order provides a clear visual hierarchy, guiding users to the most important actions first.
```

File: 1004-button-group-design-guidance.md [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+- **Should** be used when you want similar actions to be grouped together and separated from others.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:44]:
```suggestion
- Design teams **should** use button groups to organize similar actions together, clearly separating them from other actions. This grouping helps users quickly identify related options.
```

File: 1004-button-group-design-guidance.md [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+
+### Checkbox Button Groups
+
+- **Should** be used when a small list of options can be selected from, similar to the [checkbox](https://clarity.design/documentation/checkbox) component.
+- **Should** use option labels that are 1-3 short words.
+
+### Radio Button Groups
+
+- **Should** be used when selecting one option from a small list of options, similar to the [radio button](https://clarity.design/documentation/radio) component.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:53]:
```suggestion
- Design teams **should** use button groups when selecting one option from a small list, similar to the [radio button](https://clarity.design/documentation/radio) component. This ensures a clear and straightforward selection process for users.
```

File: 1004-button-group-design-guidance.md [hunk 26]
```diff
@@ -0,0 +1,58 @@
...
+- **Should** use option labels that are 1-3 short words.
```
Comment (jbusta5) [inline 1004-button-group-design-guidance.md:54]:
```suggestion
- Design teams **should** use option labels consisting of 1 to 3 short words. Keeping labels concise improves clarity and helps users quickly grasp the available actions.
```

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**